pkg/golinters/internal/diff.go

```diff
@@ -3,20 +3,23 @@ package internal
 import (
 	"bytes"
 	"fmt"
+	"go/ast"
 	"go/token"
+	"slices"
 	"strings"
 
 	diffpkg "github.com/sourcegraph/go-diff/diff"
+	"golang.org/x/tools/go/analysis"
 
 	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/goanalysis"
 	"github.com/golangci/golangci-lint/pkg/lint/linter"
 	"github.com/golangci/golangci-lint/pkg/logutils"
-	"github.com/golangci/golangci-lint/pkg/result"
 )
 
 type Change struct {
-	LineRange   result.Range
-	Replacement result.Replacement
+	From, To int
+	NewLines []string
```

Comment:
`NewText` and `NewLines` are different: one is a slice of bytes (a string), one is a slices of string.
The name `NewLines` is right because it contains lines and not the full text.